lib/ansible/plugins/connection/network_cli.py

```diff
@@ -359,6 +360,14 @@ def close(self):
                 display.debug("ssh connection has been closed successfully")
         super(Connection, self).close()
 
+    def receive_ssh_data(self, count, timeout):
+        start = time.time()
+        while timeout and self._ssh_shell.recv_ready() is False:
+            if time.time() - start >= timeout:
+                raise AnsibleConnectionFailure("timeout waiting ouput from command")
+            time.sleep(0.001)
```

Comment:
I am still not clear why the delay is required here. Can't this just be

```
while True:
  if self._ssh_shell.recv_ready():
    self._ssh_shell.recv(256)
```
The `persistent_command_timeout` timer triggered [here](https://github.com/ansible/ansible/blob/devel/bin/ansible-connection#L138) will handle breaking from loop if in case response is not received within the time set by command timeout. Unless I am missing something here the 
`raise AnsibleConnectionFailure` is not required.